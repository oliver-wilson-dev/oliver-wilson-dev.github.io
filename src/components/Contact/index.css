.contactIcons {
    line-height: 50px;
    height: 50px;
}

.contactDetail {
    display: inline-block;
    margin: 0 5px 0 auto;
}