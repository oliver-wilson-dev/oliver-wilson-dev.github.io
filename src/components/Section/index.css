@import "../../styles/transitions.css";

@keyframes fadeIn {
    0% { 
        opacity: 0; 
    }
    100% { 
        opacity: 1; 
    }
}

.section {
    transition: visibility var(--sectionVisibilityTransitionDuration), opacity var(--sectionOpacityTransitionDuration);
    transition-delay: var(--spinnerOpacityTransitionDuration);
    opacity: 0;
    visibility: hidden;
    &.show {
        opacity: 1;
        visibility: visible;
      }
}

.displayBox {
    background: var(--primaryBackground);
    -webkit-box-shadow: 8px 8px 0px var(--primaryBorderColour);
    -moz-box-shadow: 8px 8px 0px var(--primaryBorderColour);
    box-shadow: 8px 8px 0px var(--primaryBorderColour);
    border: 3px solid var(--primaryBorderColour);
    padding: 10px;
    max-height: 250px;
    overflow: hidden;
    position: relative;
}

.displayBoxExpanded {
    max-height: 1500px;
}

.collapseExpand {
    margin: 5px 5px 5px auto;
    position: absolute;
    right: 0px;
    top: 0px;
    font-size: 30px;
    width: 35px;
    border: none;
    background: transparent;
    font-family: inherit;
    font-weight: 900;
    color: inherit;
}

@media all and (min-width: 786px) {
    .landingBlurb {
        max-width: 800px;
    }
}