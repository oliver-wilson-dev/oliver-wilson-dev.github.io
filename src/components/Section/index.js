import React from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';
import styles from './index.css';
import sharedStyles from '../App/index.css';

class Section extends React.Component {
  constructor(props) {
    super(props);

    this.displayBoxRef = null;
    this.childrenRef = React.createRef();

    this.state = {
      expanded: false,
      expandable: false,
      visible: false
    };

    this.setDisplayBoxRef = this.setDisplayBoxRef.bind(this);

    this.canBeFocusedProgrammatically = -1;
  }


  componentDidMount() {
    this.setState(
      previousState => ({
        ...previousState,
        expandable: this.displayBoxRef.offsetHeight >= 250,
        visible: true
      })
    );
  }

  componentDidUpdate() {
    const { state: { expanded } } = this;

    if (expanded) { this.childrenRef.current.focus(); }
  }

  setDisplayBoxRef(elm) {
    this.displayBoxRef = elm;
  }

  onButtonClick = (e) => {
    e.preventDefault();
    this.setState(previousState => ({
      ...previousState,
      expanded: !previousState.expanded
    }));
  }

  render() {
    const {
      canBeFocusedProgrammatically,
      onButtonClick,
      setDisplayBoxRef,
      props: { title, children },
      state: { expanded, expandable, visible },
    } = this;

    return (
      <div className={cn(sharedStyles.flexColumn, sharedStyles.flexCenter, styles.section, {
        [styles.show]: visible
      })}
      >
        <h2 className={sharedStyles.sectionTitle}>{title}</h2>
        <div
          ref={setDisplayBoxRef}
          className={(cn(styles.landingBlurb, styles.displayBox,
            {
              [styles.displayBoxExpanded]: expanded,
              [styles.blurBottom]: expandable && !expanded
            }))
          }
        >
          {expandable && (
          <button
            className={styles.collapseExpand}
            type="button"
            onClick={onButtonClick}
            aria-expanded={expanded}
            aria-label={expanded ? 'click to minimise this section' : 'click to expand this section to reveal more content'}
          >
            <span>{expanded ? 'âˆ’' : '+'}</span>
          </button>
          )}
          <div
            ref={this.childrenRef}
            tabIndex={canBeFocusedProgrammatically}
            aria-hidden={expandable && !expanded}
          >
            {children}
          </div>
        </div>
      </div>
    );
  }
}

Section.propTypes = {
  title: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired
};

export default Section;
