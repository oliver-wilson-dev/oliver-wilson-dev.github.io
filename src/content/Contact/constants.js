export const EMAIL = 'oliver.wilson@asos.com';
export const PHONE_NUMBER = '07580081310';
