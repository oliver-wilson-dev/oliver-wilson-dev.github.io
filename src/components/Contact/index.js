import React from 'react';
import PropTypes from 'prop-types';
import Section from '../Section';
import styles from './index.css';

const Contact = ({ title, phone: phoneNumber, email }) => (
  <Section title={title}>
    <div aria-label={`my phone number is ${phoneNumber}`}>
      <h4 className={styles.contactDetail} aria-hidden>
        Phone:
      </h4>
      <span aria-hidden>{phoneNumber}</span>
    </div>
    <div aria-label={`my email address is ${email}`}>
      <h4 className={styles.contactDetail} aria-hidden>
        Email:
      </h4>
      <span aria-hidden>{email}</span>
    </div>
  </Section>
);

Contact.propTypes = {
  title: PropTypes.string.isRequired,
  phone: PropTypes.string.isRequired,
  email: PropTypes.string.isRequired
};

export default Contact;
