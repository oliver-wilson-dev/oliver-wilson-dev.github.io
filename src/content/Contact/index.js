import React from 'react';
import styles from '../index.css';
import { EMAIL, PHONE_NUMBER } from './constants';

export default {
  title: 'Get in touch',
  content:
  <div>
    <div>
      <h4 className={styles.contactDetail}>
        Phone:
      </h4>
      <span>{PHONE_NUMBER}</span>
    </div>
    <div>
      <h4 className={styles.contactDetail}>
        Email:
      </h4>
      <span>{EMAIL}</span>
    </div>
  </div>,
};
