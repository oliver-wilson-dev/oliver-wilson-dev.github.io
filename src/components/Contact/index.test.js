import React from 'react';
import { shallow } from 'enzyme';
import Contact from '.';
import Section from '../Section';

const defaultProps = {
  title: 'test-title',
  phone: 'test-phone-number',
  email: 'test-email'
};

const render = overrideProps => shallow(<Contact {...defaultProps} {...overrideProps} />);

jest.mock('../Section', () => {
  const Section = () => null;

  return Section;
});

describe('<Contact/> component', () => {
  it('should exist', () => {
    expect(render().exists()).toBe(true);
  });

  it('should match snapshot', () => {
    expect(render()).toMatchSnapshot();
  });

  it('should render a Section component and pass it the "title" prop', () => {
    const testTitle = 'test-title';
    expect(render({ title: testTitle }).find(Section).prop('title')).toBe(testTitle);
  });
});
