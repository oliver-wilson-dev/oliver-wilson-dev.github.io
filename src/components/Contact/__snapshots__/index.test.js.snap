// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Contact/> component should match snapshot 1`] = `
<Section
  title="test-title"
>
  <div
    aria-label="my phone number is test-phone-number"
  >
    <h4
      aria-hidden={true}
      className="contactDetail"
    >
      Phone:
    </h4>
    <span
      aria-hidden={true}
    >
      test-phone-number
    </span>
  </div>
  <div
    aria-label="my email address is test-email"
  >
    <h4
      aria-hidden={true}
      className="contactDetail"
    >
      Email:
    </h4>
    <span
      aria-hidden={true}
    >
      test-email
    </span>
  </div>
</Section>
`;
